### PR TITLE
[TACHYON-231] Add utilities for working with ByteBuffer

### PR DESCRIPTION
**Pull request by xinjin**
https://tachyon.atlassian.net/browse/TACHYON-231

### CODE REVIEW

**Comment by calvinjia** [comment]
@xinjin Thanks for the patch! I would prefer if the new util method is used in the current code as opposed to just providing it for future use. Do you mind using this method where appropriate?

**Comment by xinjin** [comment]
@calvinjia Do you mean we find all the related existing code and replace them with this util function? Do we create a new ticket and pull request for this, or do this with the current ticket and pull request?

**Comment by calvinjia** [comment]
@xinjin Yeah, it would be great if we could do that in this PR. This way we do not have unused code in the codebase.

**Comment by xinjin** [comment]
I used grep to search for 0xFF and only saw BlockOutStream used this. I have replaced the code with the new util function.

**Comment by calvinjia** [comment]
Thanks, could you resolve the merge conflicts? Overall LGTM

**Comment by xinjin** [comment]
@calvinjia Merged with the upstream.

**Comment by haoyuan** [comment]
Jenkins, test this please.

**Comment by haoyuan** [comment]
Jenkins, add to whitelist.

**Comment by haoyuan** [comment]
Jenkins, test this please.

**Comment by calvinjia** [comment]
@xinjin Could you include the method in a Utils class in the client package? Client is no longer depending on core so it cannot get core's `tachyon.io.Utils`

**Comment by haoyuan** [comment]
We can’t automatically merge this pull request.

**Comment by xinjin** [comment]
@haoyuan @calvinjia Sorry, I am traveling this week. I will fix this soon.

**Comment by xinjin** [comment]
@calvinjia I placed the function to CommonUtils which can be used by the client after the refactoring.

File: common/src/main/java/tachyon/util/CommonUtils.java [hunk 1]
```diff
@@ -360,6 +360,10 @@ public static void printTimeTakenNs(long startTimeNs, Logger logger, String mess
     logger.info(message + " took " + (System.nanoTime() - startTimeNs) + " ns.");
   }
 
+  public static void putIntByteBuffer(ByteBuffer buf, int b) {
+    buf.put((byte)(b & 0xFF));
```
Comment (calvinjia) [inline common/src/main/java/tachyon/util/CommonUtils.java]: Could this be changed to the previous style (space in between)?

**Comment by calvinjia** [comment]
@xinjin Thanks, overall LGTM, one minor comment. Also, it seems like the commit history is messy, could it be cleaned up?

**Comment by xinjin** [comment]
@calvinjia I cleaned the commits and added the space as you suggested. Nothing else changed. But the test failed on one case. What goes wrong?

**Comment by xinjin** [comment]
Jenkins, retest this please.

**Comment by calvinjia** [comment]
+1 LGTM

That test is sometimes unstable.

**Comment by apc999** [comment]
LGTM

**Comment by xinjin** [comment]
@calvinjia If it's OK, let's merge the change and close this PR.

**Comment by haoyuan** [comment]
Jenkins, test this please.

**Comment by haoyuan** [comment]
Jenkins, test this please.

**Comment by haoyuan** [comment]
Jenkins, test this please.

**Comment by haoyuan** [comment]
Thanks!

**Pull request merged**